controllers/withwatcher/kyma_controller_test.go

```diff
@@ -21,90 +27,212 @@ import (
 	expectedWebhookNamePartsLength = 4
+	dummyCaBundleData              = "amVsbHlmaXNoLXdhcy1oZXJl"
 )
 
-var _ = Describe("Kyma with multiple module CRs in remote sync mode", Ordered, func() {
-	kyma := NewTestKyma("kyma-remote-sync")
-	watcherCrForKyma := createWatcherCR("skr-webhook-manager", true)
+var (
+	ErrExpectedAtLeastOneWebhook       = errors.New("expected at least one webhook configured")
+	ErrWebhookConfigForWatcherNotFound = errors.New("webhook config matching Watcher CR not found")
+	ErrWebhookNamePartsNumberMismatch  = errors.New("webhook name dot separated parts number mismatch")
+	ErrManagedByLabelNotFound          = errors.New("managed-by label not found")
+	ErrModuleNameMismatch              = errors.New("module name mismatch")
+	ErrSvcPathMismatch                 = errors.New("service path mismatch")
+	ErrWatchLabelsMismatch             = errors.New("watch labels mismatch")
+	ErrStatusSubResourcesMismatch      = errors.New("status sub-resources mismatch")
+	ErrSpecSubResourcesMismatch        = errors.New("spec sub-resources mismatch")
+)
 
+var _ = Describe("Kyma with multiple module CRs in remote sync mode", Ordered, Serial, func() {
+	webhookConfig := &admissionv1.ValidatingWebhookConfiguration{}
+	kyma := NewTestKyma("kyma-remote-sync")
 	kyma.Spec.Sync = v1alpha1.Sync{
 		Enabled:      true,
 		Strategy:     v1alpha1.SyncStrategyLocalClient,
 		Namespace:    metav1.NamespaceDefault,
 		NoModuleCopy: true,
 	}
+	kymaObjKey := client.ObjectKeyFromObject(kyma)
+	watcherCrForKyma := createWatcherCR("skr-webhook-manager", true)
 	registerDefaultLifecycleForKymaWithWatcher(kyma, watcherCrForKyma)
 
 	It("kyma reconciler installs watcher helm chart with correct webhook config", func() {
-		webhookConfig := &admissionv1.ValidatingWebhookConfiguration{}
-		Eventually(isWebhookDeployed(suiteCtx, runtimeClient, webhookConfig), Timeout, Interval).
-			Should(Succeed())
-		Expect(isWebhookConfigured(watcherCrForKyma, webhookConfig)).To(BeTrue())
-		Eventually(IsKymaInState(suiteCtx, controlPlaneClient, kyma.GetName(), v1alpha1.StateReady),
+		By("waiting for skr webhook condition to be ready")
+		Eventually(IsSKRWebhookConditionReasonReady(suiteCtx, controlPlaneClient, kymaObjKey),
+			Timeout, Interval).Should(BeTrue())
+		By("waiting for webhook configuration to be deployed in the remote cluster")
+		Eventually(runtimeClient.Get(suiteCtx, client.ObjectKey{
+			Namespace: metav1.NamespaceDefault,
+			Name:      deploy.ResolveSKRChartResourceName(deploy.WebhookConfigNameTpl, kymaObjKey),
+		}, webhookConfig), Timeout, Interval).Should(Succeed())
+		Expect(isWebhookConfigured(watcherCrForKyma, webhookConfig, kyma.Name)).To(Succeed())
+	})
+
+	It("kyma reconciler re-installs watcher helm chart when webhook CA bundle is not consistent", func() {
+		By("updating webhook config with corrupt CA bundle data")
+		Expect(webhookConfig.Webhooks).NotTo(BeEmpty())
+		webhookConfig.Webhooks[0].ClientConfig.CABundle = []byte(dummyCaBundleData)
```

Comment:
are you updating this `dummyCaBundleData`. As far as I see it's only set on the variable and not in the testenv cluster?